### PR TITLE
Transfer current tokens to escrow when setting new token in bridgeMinter

### PR DESCRIPTION
**Pull request by kautukkundan**
**What does this pull request do? Explain your changes. (required)**

**Specific updates (required)**

- transfers tokens held by BridgeMinter to a specified address before setting a new token

**How did you test each of these updates (required)**

see commit history

**Does this pull request close any open issues?**

**Checklist:**

- [x] README and other documentation updated
- [x] All tests using `yarn test` pass

### CODE REVIEW

**Review by yondonfu — CHANGES_REQUESTED** [review]
Instead of having a `setToken()` function for changing the token at all I suggest:

- Using a [livepeerToken()](https://github.com/livepeer/protocol/blob/20e7ebb86cdb4fe9285bf5fea02eb603e5d48805/contracts/token/Minter.sol#L247) function to fetch the LPT address from the Controller
- `migrateToNewMinter()` will transfer the contract's balance to a new contract if needed so in the unlikely event that the LPT address returned by the Controller ever changes it would just be important to call `migrateToNewMinter()` first to move the contract's balance for the existing LPT address

This avoids the need for an explicit setter and dealing with the additional logic of transferring tokens in that setter right now.

The above does involve extra gas usage due to the external call to the Controller, but since the operations that involve making calls to the LPT contract are infrequent I don't think its that big of a deal.

File: contracts/token/BridgeMinter.sol [hunk 1]
```diff
@@ -135,4 +124,11 @@ contract BridgeMinter is Manager {
     function getController() public view returns (address) {
         return address(controller);
     }
+
+    /**
+     * @dev Returns IBridgeMinterToken interface
+     */
+    function livepeerToken() private view returns (IBridgeMinterToken) {
+        return IBridgeMinterToken(IController(controller).getContract(keccak256("LivepeerToken")));
```
Comment (yondonfu) [inline contracts/token/BridgeMinter.sol]:
```suggestion
        return IBridgeMinterToken(controller.getContract(keccak256("LivepeerToken")));
```

`controller` should already be a `IController` type.

**Review by yondonfu — CHANGES_REQUESTED** [review]
Looks good! Just one suggested change.

File: contracts/token/BridgeMinter.sol (hunk 1 above)
Comment (kautukkundan) [reply contracts/token/BridgeMinter.sol]: fixed in 0e00833

**Comment by kautukkundan** [comment]
rebased from target

**Review by yondonfu — APPROVED** [review]
LGTM. Let's squash and merge using a single commit that still references the relevant finding.

**Pull request merged**